### PR TITLE
CM-111: fixed getting fiat prices from min-api.cryptocompare.com

### PR DESCRIPTION
**Pull request by ozalexo**
There is API limitation: fsyms - Comma separated cryptocurrency symbols list [Max character length: 300]
Link: https://min-api.cryptocompare.com/ (tab "Multiple Symbols Price")

This PR fixing this error: 
screen shot 2018-08-23 at 15 26 03

### CODE REVIEW

File: packages/core/redux/market/utils.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// eslint-disable-next-line import/prefer-default-export
+export const chunker = (tokens) => {
+
+  const chunk = (list, chunkSize = 5) => {
+    if (!list || !list.length) {
+      return []
+    }
+
+    let i, j, t
```
Comment (wAISw) [inline packages/core/redux/market/utils.js]: improve readability

File: packages/core/redux/market/actions.js [hunk 2]
```diff
@@ -26,9 +27,20 @@ const watchMarket = (dispatch, getState) => async () => {
   if (tokens.length === 0 || currencies.length === 0) {
     return
   }
-  const response = await axios.get(`https://min-api.cryptocompare.com/data/pricemulti?fsyms=${tokens.join(',')}&tsyms=${currencies.join(',')}`)
-  const prices = response ? response.data : {}
-  dispatch({ type: MARKET_UPDATE_PRICES, prices })
+  let response = {}
+  const tokenList = tokens.join(',')
+  const currencyList = currencies.join(',')
+  if (tokenList.length > 300) {
```
Comment (e-maximus) [inline packages/core/redux/market/actions.js]: I'd make a constant for number 300

File: packages/core/redux/market/utils.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+// eslint-disable-next-line import/prefer-default-export
+export const chunker = (tokens) => {
+
+  // chunkSize is 35 by default. Lenght of a token may be 3 or 6 characters (especially in testnet)
+  // 3*35 or 6*35 = ~200 or ~300 characters in comma separated string
+  const chunk = (list, requiredChunkSize = 35) => {
```
Comment (e-maximus) [inline packages/core/redux/market/utils.js]: Also I'd put 35 into the constant

File: packages/core/redux/market/actions.js [hunk 4]
```diff
@@ -26,9 +27,20 @@ const watchMarket = (dispatch, getState) => async () => {
...
+    const chunks = chunker(tokens)
+    const chunkedResponse = await Promise.all(
+      chunks.map((chunk) => axios.get(`https://min-api.cryptocompare.com/data/pricemulti?fsyms=${chunk.join(',')}&tsyms=${currencyList}`))
+    )
+    response['data'] = {}
+    chunkedResponse.forEach((cResponse) => response.data = { ...response.data, ...cResponse.data })
+  } else {
+    response = await axios.get(`https://min-api.cryptocompare.com/data/pricemulti?fsyms=${tokenList}&tsyms=${currencyList}`)
+  }
+  Object.keys(response.data).length && dispatch({ type: MARKET_UPDATE_PRICES, prices: response.data })
```
Comment (e-maximus) [inline packages/core/redux/market/actions.js]: Maybe do better with classical if

```
if (Object.keys(response.data).length) {
  dispatch({ type: MARKET_UPDATE_PRICES, prices: response.data })
}
```

**Pull request merged**